dask/array/_shuffle.py

```diff
@@ -294,8 +294,30 @@ def _getitem(obj, index):
 
 
 def concatenate_arrays(arrs, sorter, axis):
-    concatenate = concatenate_lookup.dispatch(type(arrs[0]))
-    return np.take(concatenate(arrs, axis=axis), np.argsort(sorter[1]), axis=axis)
+    try:
```

Comment:
You have to hook this into our dispatching mechanism, cc @rjzamora who will probably be able to provide guidance here

Reply:
Thanks @phofl - yeah, I'm not exactly sure what dispatching mechanism you're referring to.  If you want me to create one for indexing, I can (like `concatenate_lookup` and `tensordot_lookup`)

Reply:
Yeah we basically need something like ``concatenate_lookup`` for this

Reply:
Great will do

Reply:
maybe a better name than ``concatenate_arrays`` if you end up defining the function outside of ``_shuffle.py``

Reply:
I guess what I'm saying is that something like a `concatenate_and_take` registray would need to some include a way of dispatching the `take` which is what I do here.  But we wouldn't need a registry for that then.  I'll think on it some more.

Reply:
As far as I can tell, the `take_lookup` solution makes sense. It's probably best to dispatch on primitive logic that is likely to be used in many places (e.g. concat and take).